code/modules/recycling/disposal.dm

```diff
@@ -615,12 +614,13 @@
 				H.take_overall_damage(20, 0, "Blunt Trauma")//horribly maim any living creature jumping down disposals.  c'est la vie
 				*/
 
-		if(has_fat_guy && prob(2)) // chance of becoming stuck per segment if contains a fat guy
-			active = 0
+		if(prob(2)) // chance of becoming stuck per segment if contains a fat guy
+			if(has_fat_guy())
+				active = 0
 			// find the fat guys
-			for(var/mob/living/carbon/human/H in src)
+				for(var/mob/living/carbon/human/H in src)
 
-			break
+				break
```

Comment:
Won't this break out of the while loop?

Reply:
That's what it does now. If I took it out then it would change the behavior to have an extra step, right? You wouldn't be able to get stuck in a one segment pipe. But that for statement above it is just empty, yeah. I'll remove that.